### PR TITLE
feat: multi-lines

### PR DESCRIPTION
**Pull request by ldez**
Basic multi-lines for unambiguous elements.

### CODE REVIEW

**Comment by ldez** [comment]
@nicorikken @mojavelinux Is that kind of behavior acceptable to you?

```adoc
foobar

footnoteref:[id, text text
text text text]

foobar

footnoteref:[id, text text
text text
text

foobar
```

capture du 2016-05-12 19-56-31

File: grammars/language-asciidoc.cson [hunk 1]
```diff
@@ -471,17 +471,17 @@ repository:
     patterns: [
       {
         name: "markup.other.footnote.asciidoc"
-        match: "(?<!\\\\)(footnote(?:ref)?):\\[(.*?[^\\\\])\\]"
```
Comment (nicorikken) [inline grammars/language-asciidoc.cson]: Why did you exchange the lookbehind for a match?

**Comment by nicorikken** [comment]
Somewhat better, assuming people don't make mistakes like leaving out the closing brackets. But that would better be handled using a linter.
Some of my edge case testing:
screen shot 2016-05-12 at 20 04 15
```asciidoc
foobar

footnoteref:[id, text text
text text text]

foobar

footnoteref:[id, text text
text text
text

foobar
```

I think this is still better than the single-line variant, assuming valid Asciidoctor syntax.

File: grammars/language-asciidoc.cson (hunk 1 above)
Comment (ldez) [inline grammars/language-asciidoc.cson]: bad copy paste. fix

**Comment by ldez** [comment]
If you assuming this is possible, I apply this to:
- footnote(ref)
- passthrough (only the `pass:[text]` variant)
- stem
- xref

**Comment by nicorikken** [comment]
Certainly I think you can add this to the other constructs with a similar (bounded) syntax. That leave only the more conflicting constructs up for debate.

**Comment by ldez** [comment]
I can do that:

capture du 2016-05-12 21-16-12

Is this expected behavior?

**Comment by ldez** [comment]
It's ready to review.

Say me if spaces at the ending of line are invalid characters.

**Comment by ldez** [comment]
I have also fixed all macros for respect naming conventions and styling.

**Comment by nicorikken** [comment]
I see no breakage in practical coloring (due to renaming). Code is nice. Spaces at the end seem to render neatly in Asciidoc preview plugin, so would be valid. As long as the macro is closed.

This can be merged.

**Comment by mojavelinux** [comment]
> Say me if spaces at the ending of line are invalid characters.

The first thing the converter does is strip all trailing whitespace. So technically it is valid because it is ignored. But we could highlight it as an error to show that it will get stripped away and is thus superfluous.

**Comment by mojavelinux** [comment]
> Somewhat better, assuming people don't make mistakes like leaving out the closing brackets. But that would better be handled using a linter.

True. It's also unavoidable given the current behavior of first-mate. They really, really, really need to add a way to validate the inner content against a spec. Right now, this is the very best we can do.

**Pull request merged**